### PR TITLE
[IAST] Add support for invoke dynamic csi advices

### PR DESCRIPTION
**Pull request by manuel-alvarez-alvarez**
# What Does This Do
Add support for Invoke Dynamic instructions to the Call Site instrumenter

# Motivation
From JDKs >= 9 the solution for `+` concatenation is based on invoke dynamic instructions instead of the traditional `StringBuilder` approach

### CODE REVIEW

**Review by anderruiz — APPROVED** [review]

File: dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/csi/Advices.java [hunk 1]
```diff
@@ -187,14 +212,16 @@ class ConstantPoolInstrospector implements AdviceIntrospector {
 
       private static final int CONSTANT_METHOD_REF = 10;
       private static final int CONSTANT_INTERFACE_METHOD_REF = 11;
+      private static final int CONSTANT_INVOKE_DYNAMIC = 18;
 
       private static final Map<Integer, ConstantPoolHandler> CP_HANDLERS;
 
       static {
-        final Map<Integer, ConstantPoolHandler> handlers = new HashMap<>(2);
+        final Map<Integer, ConstantPoolHandler> handlers = new HashMap<>(3);
```
[the remark this answers is not in the document]
Comment (manuel-alvarez-alvarez) [reply dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/csi/Advices.java]: Yep, thanks!. Never checked the actual code at `HashMap`
```
    /**
     * Returns a power of two size for the given target capacity.
     */
    static final int tableSizeFor(int cap) {
        int n = -1 >>> Integer.numberOfLeadingZeros(cap - 1);
        return (n < 0) ? 1 : (n >= MAXIMUM_CAPACITY) ? MAXIMUM_CAPACITY : n + 1;
    }
```

File: dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/csi/CallSiteTransformer.java [hunk 2]
```diff
@@ -48,6 +51,9 @@ private CallSiteVisitorWrapper(@Nonnull final Advices advices) {
 
     @Override
     public int mergeWriter(final int flags) {
+      if (!advices.computeMaxStack()) {
```
[the remark this answers is not in the document]
Comment (manuel-alvarez-alvarez) [reply dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/csi/CallSiteTransformer.java]: Yes you are right, negations are often harder to read. Thanks!

**Review by bantonsson — APPROVED** [review]
Finally made it to the end 😅

**Comment by manuel-alvarez-alvarez** [comment]
> Finally made it to the end sweat_smile

This one was dense, thanks!!!

**Pull request merged**